### PR TITLE
Up our CI game

### PR DESCRIPTION
**Pull request by sideeffffect**
Analogous PR in ZIO Prelude https://github.com/zio/zio-prelude/pull/592

 * use Renovate to maintain versions in GitHub Actions
 * manually managing each CI sub-step is unsustainable, now we have a step `ci` which aggregates all the necessary steps and which other tools can reference (e.g. GitHub or Mergify)
 * single source of truths for versions
   * Scala versions are read from `.github/workflows/ci.yml`
   * in `.github/workflows/ci.yml`, versions are not duplicated
 * better policy for Release Drafter, with better labeling support (taken from Release Drafter itself)

### CODE REVIEW

**Comment by sideeffffect** [comment]
@mijicd @adamgfraser Could you please mark `ci`, `license/cla` and `Summary` as the only Required build steps in `master` branch protection settings?

We will probably also need to activate Renovate for ZIO (and ZIO Prelude) after this PR gets merged.
https://app.renovatebot.com/dashboard

**Comment by adamgfraser** [comment]
@mijicd I am assuming you have this one but let me know if you need anything from me.

**Review by mijicd — APPROVED** [review]
As I mentioned in the related zio-prelude PR, I'm in favor of this change. Nevertheless, I want to restate the importance of propagating it to all repos in the organization (including the ones that aren't on actions yet).

Also, before merging this, I'd like to get "yay" or "nay" from @softinio.

P.S. Assuming it lands, I'll set up the recommended protection and enable the integration.

**Comment by sideeffffect** [comment]
Dotty build seemed to have failed only on a flaky test :smile: Could somebody restart it, please?

File: .github/workflows/ci.yml
Comment (sideeffffect) [inline .github/workflows/ci.yml:11]: @mijicd just checking... all those options need to be there, `-Xmx` wouldn't be enough, right?

Comment (mijicd) [reply .github/workflows/ci.yml:11]: They can probably be pruned, as they were ported from CircleCI setup. However, note that they were introduced there using the good old trial and error :).

File: project/BuildHelper.scala
Comment (fsvehla) [inline project/BuildHelper.scala:11]: love this!

File: .github/workflows/ci.yml
Comment (sideeffffect) [reply .github/workflows/ci.yml:11]: So I've kept only `-Xmx` and it seems to be working OK :crossed_fingers:

Comment (softinio) [reply .github/workflows/ci.yml:11]: Thanks for optimizing this as long as we maintain pipeline stability I am happy. Completely agree with keeping the `-Xmx` option as that made the difference when I was tweaking this last.

**Review by softinio — APPROVED** [review]
Overall happy to proceed with these improvements. Further to my other comments it would be preferred if we standardized on all other repos so a plan to spread the love to other repos (of course with other contributors and maintainers help) would be appreciated. 

Thanks @mijicd for including me in this conversation.

**Review by mijicd — APPROVED** [review]

**Pull request merged**
